packages/opentelemetry-plugin-grpc-js/src/grpcJs.ts

```diff
@@ -14,32 +14,75 @@
  * limitations under the License.
  */
 
+import type * as grpcJs from '@grpc/grpc-js';
 import { BasePlugin } from '@opentelemetry/core';
+import * as shimmer from 'shimmer';
+import { patchClient, patchLoadPackageDefinition } from './client';
+import { patchServer } from './server';
 import { VERSION } from './version';
-import * as path from 'path';
-
-import * as grpcJs from '@grpc/grpc-js';
 
 /**
  * @grpc/grpc-js gRPC instrumentation plugin for Opentelemetry
  * https://www.npmjs.com/package/@grpc/grpc-js
  */
 export class GrpcJsPlugin extends BasePlugin<typeof grpcJs> {
   static readonly component = '@grpc/grpc-js';
+
   readonly supportedVersions = ['1.*'];
 
-  constructor(readonly moduleName: string, readonly version: string) {
+  tracer = this._tracer;
+
+  logger = this._logger;
+
+  constructor(readonly moduleName: string) {
     super('@opentelemetry/plugin-grpc-js', VERSION);
   }
 
   protected patch(): typeof grpcJs {
-    throw new Error('Method not implemented.');
+    this.tracer = this._tracer;
```

Comment:
should this be somehow unpatched in `unpatch`

Reply:
Good point. This would be to prevent a dangling reference on this reference right? I've added jsdoc comment to their declarations and nulled them out here via
```ts
(this.tracer as Tracer | null) = null;
(this.logger ...) = null
```

Reply:
I don't think a dangling reference here is a big deal as there can only ever be one of them. If there was a new reference on each call to patch then maybe it would be a problem.